aws/resource_aws_s3control_bucket_test.go

```diff
@@ -0,0 +1,234 @@
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSS3ControlBucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSS3ControlBucketConfig_Bucket(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3ControlBucketExists(resourceName),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsS3ControlBucket(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSS3ControlBucket_Tags(t *testing.T) {
+	TestAccSkip(t, "S3 Control Bucket resource tagging requires additional eventual consistency handling, see also: https://github.com/terraform-providers/terraform-provider-aws/issues/15572")
+
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_s3control_bucket.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t); testAccPreCheckAWSOutpostsOutposts(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSS3ControlBucketDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSS3ControlBucketConfig_Tags1(rName, "key1", "value1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3ControlBucketExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSS3ControlBucketConfig_Tags2(rName, "key1", "value1updated", "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3ControlBucketExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1updated"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+			{
+				Config: testAccAWSS3ControlBucketConfig_Tags1(rName, "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSS3ControlBucketExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSS3ControlBucketDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).s3controlconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_s3control_bucket" {
+			continue
+		}
+
+		parsedArn, err := arn.Parse(rs.Primary.ID)
+
+		if err != nil {
+			return fmt.Errorf("error parsing S3 Control Bucket ARN (%s): %w", rs.Primary.ID, err)
+		}
+
+		input := &s3control.GetBucketInput{
+			AccountId: aws.String(parsedArn.AccountID),
+			Bucket:    aws.String(rs.Primary.ID),
+		}
+
+		_, err = conn.GetBucket(input)
+
+		if tfawserr.ErrCodeEquals(err, "NoSuchBucket") {
+			return nil
```

Comment:
should we continue through the for-loop here in case? though I've seen it written both ways where we return `nil` right away instead 🤔

Reply:
hehe just peeped https://github.com/terraform-providers/terraform-provider-aws/issues/15847 👍

Reply:
Ha yeah, should be `continue`, will suggest fix and merge